addon/mixins/run.js

```diff
@@ -88,18 +89,15 @@ export default Mixin.create({
       !this.isDestroyed
     );
 
-    let pendingTimers = getOrAllocate(this, '_pendingTimers', Array);
-
     let cancelId = run.later(() => {
-      let cancelIndex = pendingTimers.indexOf(cancelId);
-      pendingTimers.splice(cancelIndex, 1);
+      getOrAllocate(this, '_registeredDisposables', Map).delete(cancelId);
 
       let task = getTask(this, taskOrName, 'runTask');
 
       task.call(this);
     }, timeout);
 
-    pendingTimers.push(cancelId);
+    registerDisposable(this, cancelId, () => run.cancel(cancelId));
```

Comment:
I don't think we want a callback registered per task run, per instance. I would like to avoid the overhead of all those extra functions.

We should:

* continue to build up the array per task run
* register a single "cleanup" per instance, that handles all cleanup.

Reply:
I hear you. There effectively is a single cleanup - the map is iterated and cancels are scheduled - but there definitely are multiple callbacks. Are you proposing that we just capture the `cancelId`s, and merely iterate/run.cancel all at once?

Reply:
> Are you proposing that we just capture the cancelIds, and merely iterate/run.cancel all at once?

If possible yes, that is unless there is a constraint I am missing.

Reply:
I'm leaning towards agreeing with you. It _feels_ wrong to be creating so many atomic functions that essentially do the same thing. I'm going to explore other approaches to simplify.